### PR TITLE
uom data conversion explanation and queries

### LINKED ISSUE #1034: Migration scripts for new units and measures model

### PR DESCRIPTION
**Pull request by philblackwood**
Unit of Measure data conversion steps (powerpoint) and SPARQL. Fixes #1034.

### CODE REVIEW

**Comment by mkumba** [comment]
I don't think I know how to do a git review and or approval, or if I know what I'd be looking at.  There appears to be a lot to look at.



On Apr 2, 2024, at 9:18 AM, Phil Blackwood ***@***.***> wrote:



@philblackwood<https://github.com/philblackwood> requested your review on: #1071<https://github.com/semanticarts/gist/pull/1071> uom data conversion explanation and queries.

**Comment by rjyounes** [comment]
You can make comments or modifications in a copy and then upload it into a GitHub comment.

From: Michael Uschold ***@***.***>
Date: Tuesday, April 2, 2024 at 3:27 PM
To: semanticarts/gist ***@***.***>
Cc: Rebecca Younes ***@***.***>, Mention ***@***.***>
Subject: Re: [semanticarts/gist] uom data conversion explanation and queries (PR #1071)

@philblackwood<https://github.com/philblackwood> @rjyounes<https://github.com/rjyounes> I want to make comments in the PowerPoint deck, what is the best way for me to do that? One way is to make a separate copy and email it to Phil who can make any desired changes and then make another commit. How can I do it more publicly on git?

**Review by rjyounes — APPROVED** [review]
I haven't done a thorough review of the queries but I assume everything works.

**Review by kchastain703 — APPROVED** [review]
This looks good. Thank you for your hard work!

File: migration/v13.0/queries/prepareClientReferenceData-template.rq [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# 
+# TIP: if the query becomes too large, split into sections using union to avoid cross-products
+
+# select distinct ?aspect ?unitGroup ?unitOfMeasure 
+
+construct {
+   ?broaderAspect ?broaderAspectPredicate ?broaderAspectObject .
+   ?unitGroup     ?unitGroupPredicate     ?unitGroupObject .
+   ?unitOfMeasure ?unitOfMeasurePredicate ?unitOfMeasureObject .
+  }
```
Comment (uscholdm) [inline migration/v13.0/queries/prepareClientReferenceData-template.rq:12]: An example would be super useful here. I found it difficult to make one up, in part because of the abstraction. 
- broaderAspectPredicate
- unitGroupPredicate
- unitOfMeasurePredicate

i would not epect these to vary, so much easier to just put them in.

File: migration/v13.0/uomDataConversion.pptx
Comment (uscholdm) [inline migration/v13.0/uomDataConversion.pptx:1]: [uomDataConversion-MU.pptx](https://github.com/user-attachments/files/15745182/uomDataConversion-MU.pptx)

This is a copy of the deck with comments and a few added slides.

**Review by uscholdm — CHANGES_REQUESTED** [review]
Very impressive overall, and also fairly complicated. and hard to follow. Would be much easier if there were examples.  Has every one of the queries been road tested? 

Also, the lack of availability of the unit ref data makes it more or less impossible for anyone to do the conversion.

File: migration/v13.0/uomDataConversion.pptx
Comment (philblackwood) [reply migration/v13.0/uomDataConversion.pptx:1]: Updates included: 
Remove slide showing SA internal process steps of updating, validating, and extracting subset of reference data for client.
Clarify that client reference data is not the full data reference data set, only the subset used by the client.
Improve diagrams (":" before Area and remove diagram using hasAspect).
Remove double arrows from mapping between SA reference data and client existing data.

Complete model is not included because it is not required to do the conversion.

Comment (uscholdm) [reply migration/v13.0/uomDataConversion.pptx:1]: > Complete model is not included because it is not required to do the conversion.

ok

Comment (uscholdm) [reply migration/v13.0/uomDataConversion.pptx:1]: > Improve diagrams (":" before Area and remove diagram using hasAspect).

There is no ':' in the current copy as of commit:

Commit: 254f98139fdc0d8a0ceb0e12c48d5fda6e5dd760 [254f981]
Parents: 28275d3ffa
Author: philblackwood <phil.blackwood@semanticarts.com>
Date: Monday, 10 June, 2024 10:50:23
Committer: philblackwood

Comment (uscholdm) [reply migration/v13.0/uomDataConversion.pptx:1]: > Remove double arrows from mapping between SA reference data and client existing data.

The double arrows are still there. Perhaps you made changes but they did not make it to the commit?

Comment (philblackwood) [reply migration/v13.0/uomDataConversion.pptx:1]: changes committed and pushed

Comment (uscholdm) [inline migration/v13.0/uomDataConversion.pptx:1]: Phil and I have had several go-rounds on this and it keeps improving to the point where we are mostly just fine-tuning.

**Review by uscholdm — APPROVED** [review]
